app_dart/test/request_handlers/scheduler/batch_backfiller_test.dart

```diff
@@ -0,0 +1,99 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:cocoon_service/cocoon_service.dart';
+import 'package:cocoon_service/src/model/appengine/commit.dart';
+import 'package:cocoon_service/src/model/appengine/task.dart';
+import 'package:test/test.dart';
+
+import '../../src/datastore/fake_config.dart';
+import '../../src/datastore/fake_datastore.dart';
+import '../../src/request_handling/fake_pubsub.dart';
+import '../../src/request_handling/request_handler_tester.dart';
+import '../../src/service/fake_luci_build_service.dart';
+import '../../src/service/fake_scheduler.dart';
+import '../../src/utilities/entity_generators.dart';
+
+List<Task> allGray = <Task>[
+  generateTask(1, name: 'Linux_android A', status: Task.statusNew),
+  generateTask(2, name: 'Linux_android A', status: Task.statusNew),
+  generateTask(3, name: 'Linux_android A', status: Task.statusNew),
+];
+
+List<Task> allGreen = <Task>[
+  generateTask(1, name: 'Linux_android A', status: Task.statusSucceeded),
+  generateTask(2, name: 'Linux_android A', status: Task.statusSucceeded),
+  generateTask(3, name: 'Linux_android A', status: Task.statusSucceeded),
+];
+
+List<Task> middleTaskInProgress = <Task>[
+  generateTask(1, name: 'Linux_android A', status: Task.statusNew),
+  generateTask(2, name: 'Linux_android A', status: Task.statusInProgress),
+  generateTask(3, name: 'Linux_android A', status: Task.statusNew),
+];
+
+List<Task> oldestGray = <Task>[
+  generateTask(1, name: 'Linux_android A', status: Task.statusSucceeded),
+  generateTask(2, name: 'Linux_android A', status: Task.statusSucceeded),
+  generateTask(3, name: 'Linux_android A', status: Task.statusNew),
+];
+
+final List<Commit> commits = <Commit>[
+  generateCommit(3),
+  generateCommit(2),
+  generateCommit(1),
+];
+
+void main() {
+  late BatchBackfiller handler;
+  late RequestHandlerTester tester;
+  late FakeDatastoreDB db;
+  late FakePubSub pubsub;
+  late FakeScheduler scheduler;
+
+  group('BatchBackfiller', () {
```

Comment:
could you add tests for batch scheduling cases: more than 1 target are scheduled when they all need to schedule; only one target is scheduled when itself needs to schedule, etc?

Reply:
Thanks, good point on having multiple columns being tested at a time.

To clarify, I added tests for when:
1. Only one column needs to be backfilled
2. 2 columns need to be backfilled